core/clipboard/block_paster.ts

```diff
@@ -22,15 +22,17 @@ export class BlockPaster implements IPaster<BlockCopyData, BlockSvg> {
   ): BlockSvg | null {
     if (!workspace.isCapacityAvailable(copyData.typeCounts!)) return null;
 
-    const state = copyData.saveInfo as State;
     if (coordinate) {
-      state['x'] = coordinate.x;
-      state['y'] = coordinate.y;
+      copyData.saveInfo['x'] = coordinate.x;
+      copyData.saveInfo['y'] = coordinate.y;
     }
-    return append(state, workspace, {recordUndo: true}) as BlockSvg;
+    return append(copyData.saveInfo, workspace, {recordUndo: true}) as BlockSvg;
   }
 }
 
-export interface BlockCopyData extends ICopyData {}
+export interface BlockCopyData extends ICopyData {
+  saveInfo: State;
```

Comment:
Is this keeping the name `saveInfo` just for backwards-compatibility?  Seems like a good opportunity to rename it in each sub-interface to be more descriptive.

(Here and in `workspace_comment_paster.ts`.)

Reply:
Sounds good to me!